plugins/woocommerce-admin/client/order-attribution/use-order-attribution-install-banner.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import {
+	OPTIONS_STORE_NAME,
+	PLUGINS_STORE_NAME,
+	useUser,
+} from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+
+const OPTION_NAME_BANNER_DISMISSED =
+	'woocommerce_order_attribution_install_banner_dismissed';
+const OPTION_VALUE_YES = 'yes';
+
+const useOrderAttributionInstallBanner = () => {
+	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
+	const { currentUserCan } = useUser();
+
+	const dismissOrderAttributionInstallBanner = () => {
+		updateOptions( {
+			[ OPTION_NAME_BANNER_DISMISSED ]: OPTION_VALUE_YES,
+		} );
+		recordEvent( 'order_attribution_install_banner_dismissed' );
+	};
+
+	const { canUserInstallPlugins, orderAttributionInstallState } = useSelect(
+		( select ) => {
+			const { getPluginInstallState } = select( PLUGINS_STORE_NAME );
+			const installState = getPluginInstallState(
+				'woocommerce-analytics'
+			);
+
+			return {
+				orderAttributionInstallState: installState,
+				canUserInstallPlugins: currentUserCan( 'install_plugins' ),
+			};
+		},
+		[ currentUserCan ]
+	);
+
+	const { loading, isBannerDismissed } = useSelect( ( select ) => {
+		const { getOption, hasFinishedResolution } =
+			select( OPTIONS_STORE_NAME );
+
+		return {
+			loading: ! hasFinishedResolution( 'getOption', [
+				OPTION_NAME_BANNER_DISMISSED,
+			] ),
+			isBannerDismissed: getOption( OPTION_NAME_BANNER_DISMISSED ),
+		};
+	}, [] );
+
+	return {
+		loading,
+		isOrderAttributionInstallBannerDismissed:
+			isBannerDismissed === OPTION_VALUE_YES,
+		dismissOrderAttributionInstallBanner,
+		shouldShowBanner:
```

Comment:
Are we using `isOrderAttributionInstallBannerDismissed` for something else than showing the banner or not? 

Just wondering if `isOrderAttributionInstallBannerDismissed` can be out of the exposed props and just part of the `shouldShowBanner` conditional.

Reply:
Yeah, this banner (mZgTgnHhRdyff3Gy5HIwcB-fi-2132_64200) will only be shown when the main banner is dismissed by the merchant.